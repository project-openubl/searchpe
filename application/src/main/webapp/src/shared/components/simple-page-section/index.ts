export * from "./simple-page-section";
