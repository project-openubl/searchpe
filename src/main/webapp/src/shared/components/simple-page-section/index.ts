export { SimplePageSection } from "./simple-page-section";
