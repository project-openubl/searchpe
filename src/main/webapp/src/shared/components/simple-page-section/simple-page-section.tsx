import React from "react";
import {
  PageSection,
  PageSectionVariants,
  TextContent,
  Text,
} from "@patternfly/react-core";

export interface SimplePageSectionProps {
  title: string;
  description?: string;
}

export const SimplePageSection: React.FC<SimplePageSectionProps> = ({
  title,
  description,
}) => {
  return (
    <PageSection variant={PageSectionVariants.light}>
      <TextContent>
        <Text component="h1">{title}</Text>
        {description && <Text component="small">{description}</Text>}
      </TextContent>
    </PageSection>
  );
};
