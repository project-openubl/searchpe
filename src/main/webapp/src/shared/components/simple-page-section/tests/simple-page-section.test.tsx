import React from "react";
import { shallow, mount } from "enzyme";
import { Text } from "@patternfly/react-core";
import { SimplePageSection } from "../simple-page-section";

describe("SimplePageSection", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(<SimplePageSection title="myTitle" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders without description", () => {
    const wrapper = mount(<SimplePageSection title="myTitle" />);

    expect(wrapper.find(Text).length).toEqual(1);
    expect(wrapper.find(Text).at(0).text()).toMatch(/myTitle/);
  });

  it("Renders with description", () => {
    const wrapper = mount(
      <SimplePageSection title="myTitle" description="myDescription" />
    );

    expect(wrapper.find(Text).length).toEqual(2);
    expect(wrapper.find(Text).at(0).text()).toMatch(/myTitle/);
    expect(wrapper.find(Text).at(1).text()).toMatch(/myDescription/);
  });
});
