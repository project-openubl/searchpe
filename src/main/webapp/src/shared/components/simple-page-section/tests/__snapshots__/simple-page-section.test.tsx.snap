// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimplePageSection Renders without crashing 1`] = `
<PageSection
  variant="light"
>
  <TextContent>
    <Text
      component="h1"
    >
      myTitle
    </Text>
  </TextContent>
</PageSection>
`;
